src/core/ai/llm/messages/formatters/vercel.ts

```diff
@@ -56,10 +55,12 @@ export class VercelMessageFormatter implements IMessageFormatter {
                 logger.warn(
                     'No model context available for Vercel formatter, applying conservative file filtering'
                 );
-                filteredHistory = filterMessagesByCapabilities([...history], 'groq');
+                filteredHistory = filterMessagesByLLMCapabilities([...history], {
+                    provider: 'groq',
```

Comment:
why are we using groq?

there shouldn't be an else case here, context.provider and context.model have to be mandatory, otherwise we should throw an error. defaulting to other providers would cause weird behaviours

we should make provider and model non-optional in `FormatterContext` type and any other new types introduced that include provider and model. this would be like a pre-condition for any executions